src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -216,97 +220,104 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
             }
+        }
 
-            return result;
+        private async Task<ImmutableArray<CodeFix>> GetCodeFixesAsync(
+            Document document, TextSpan span, CodeFixProvider fixer,
+            ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            var fixes = ArrayBuilder<CodeFix>.GetInstance();
+            var context = new CodeFixContext(document, span, diagnostics,
+                // TODO: Can we share code between similar lambdas that we pass to this API in BatchFixAllProvider.cs, CodeFixService.cs and CodeRefactoringService.cs?
+                (action, applicableDiagnostics) =>
+                {
+                    // Serialize access for thread safety - we don't know what thread the fix provider will call this delegate from.
+                    lock (fixes)
+                    {
+                        fixes.Add(new CodeFix(document.Project, action, applicableDiagnostics));
+                    }
+                },
+                verifyArguments: false,
+                cancellationToken: cancellationToken);
+
+            var task = fixer.RegisterCodeFixesAsync(context) ?? SpecializedTasks.EmptyTask;
+            await task.ConfigureAwait(false);
+            return fixes.ToImmutableAndFree();
         }
 
-        private async Task<List<CodeFixCollection>> AppendSuppressionsAsync(
-            Document document, TextSpan span, IEnumerable<DiagnosticData> diagnostics, List<CodeFixCollection> result, CancellationToken cancellationToken)
+        private async Task AppendSuppressionsAsync(
+            Document document, TextSpan span, IEnumerable<DiagnosticData> diagnostics, 
+            ArrayBuilder<CodeFixCollection> result, CancellationToken cancellationToken)
         {
             Lazy<ISuppressionFixProvider> lazySuppressionProvider;
             if (!_suppressionProvidersMap.TryGetValue(document.Project.Language, out lazySuppressionProvider) || lazySuppressionProvider.Value == null)
             {
-                return result;
+                return;
             }
 
-            Func<Diagnostic, bool> hasFix = (d) => lazySuppressionProvider.Value.CanBeSuppressedOrUnsuppressed(d);
-            Func<ImmutableArray<Diagnostic>, Task<IEnumerable<CodeFix>>> getFixes = (dxs) => lazySuppressionProvider.Value.GetSuppressionsAsync(document, span, dxs, cancellationToken);
-            await AppendFixesOrSuppressionsAsync(document, span, diagnostics, result, lazySuppressionProvider.Value, hasFix, getFixes, cancellationToken).ConfigureAwait(false);
-            return result;
+            await AppendFixesOrSuppressionsAsync(
+                document, span, diagnostics, result, lazySuppressionProvider.Value, 
+                hasFix: d => lazySuppressionProvider.Value.CanBeSuppressedOrUnsuppressed(d),
+                getFixes: dxs => lazySuppressionProvider.Value.GetSuppressionsAsync(
```

Comment:
No.   Not needed.   And, IMO, i avoid 'async' lambdas as i find them super confusing :)